test/e2e/main_test.go

```diff
@@ -94,10 +95,16 @@ func testMain(m *testing.M) error {
 			return false, nil
 		}
 
-		if v != 2 {
-			loopErr = fmt.Errorf("expected 2 Prometheus instances but got: %v", v)
+		i, _ := f.OperatorClient.GetInfrastructure(ctx, "cluster")
```

Comment:
you still want to handle the error?

Reply:
yes, maybe v = 1 but its not single node then we want to catch the error

Reply:
better check the error, or 0 == 0 may happen ;)

Reply:
+1 with Haoyu's comments :D

Reply:
fixed